x-pack/platform/plugins/shared/onechat/public/application/components/tools/form/sections/configuration.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut, EuiFormRow, EuiSelect } from '@elastic/eui';
+import { ToolType } from '@kbn/onechat-common/tools/definition';
+import React from 'react';
+import { Controller, useFormContext, useWatch } from 'react-hook-form';
+import { ToolFormSection } from '../components/tool_form_section';
+import { i18nMessages } from '../i18n';
+import type { ToolFormData } from '../types/tool_form_types';
+import { EsqlConfiguration } from './configuration_fields/esql_configuration_fields';
+
+export const Configuration = () => {
+  const {
+    formState: { errors },
+    control,
+  } = useFormContext<ToolFormData>();
+  const type = useWatch({ control, name: 'type' });
+
+  let configurationFields: React.ReactNode;
+  switch (type) {
```

Comment:
hmm I see a couple of switches, one in configuration (here) and 2 in Tool component

I wonder if it would be cleanre to create a "constant" file that woul ddefine per-tool type properties e..g

would it be easier to add support for new tools that way?

```
const SUPPROTED_TOOLS = {
 [ToolType.esql]: {
  configuration: () => <EsqlConfiguration />
  saveAction: async (...) => await ...
  updateAction: async (...) => await ..},
[ToolType.index_search]: {..}
}

wdyt?